code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,213 @@
+     * @return An instance of [AepDismissButton].
+     */
+    private fun createAepDismissButton(map: Map<String, Any>): AepIcon? {
+        val style = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.STYLE, MessagingConstants.ContentCard.UIKeys.NONE)
+        return when (style) {
+            MessagingConstants.ContentCard.UIKeys.SIMPLE -> AepIcon(R.drawable.close_filled)
+            MessagingConstants.ContentCard.UIKeys.CIRCLE -> AepIcon(R.drawable.cancel_filled)
+            else -> null
+        }
+    }
+
+    /**
+     * Parses the given [ContentCardSchemaData] to create an [AepUITemplate].
+     *
+     * @param contentCardSchemaData The content card schema data containing the content map.
+     * @return The parsed [AepUITemplate] or null if parsing fails.
+     */
+    internal fun getTemplateModel(contentCardSchemaData: ContentCardSchemaData): AepUITemplate? {
+        try {
+            val contentMap =
+                contentCardSchemaData.content as? Map<String, Any> ?: throw IllegalArgumentException("Content map is null")
+            val id = contentCardSchemaData.parent.proposition.uniqueId
+
+            val title = DataReader.getTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.TITLE)?.let {
+                createAepText(it)
+            }
+            if (title == null || title.content.isEmpty()) {
+                Log.error(
+                    MessagingConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Title is missing in the content card with id: $id. Skipping the content card."
+                )
+                return null
+            }
+            val body = DataReader.optTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.BODY, null)?.let {
+                createAepText(it)
+            }
+            val image = DataReader.optTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.IMAGE, null)?.let {
+                createAepImage(it)
+            }
+            val actionUrl = DataReader.optString(contentMap, MessagingConstants.ContentCard.UIKeys.ACTION_URL, null)
+            val buttons = DataReader.optTypedListOfMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.BUTTONS, emptyList()).map {
+                createAepButton(it)
+            }
+            val dismissBtn = DataReader.optTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.DISMISS_BTN, null)?.let {
+                createAepDismissButton(it)
+            }
+
+            val metaAdobeObject =
+                contentCardSchemaData.meta?.get(MessagingConstants.ContentCard.UIKeys.ADOBE)
+            metaAdobeObject.let {
+                val templateType =
+                    (it as? Map<String, Any>)?.get(MessagingConstants.ContentCard.UIKeys.TEMPLATE)
+                when (templateType) {
+                    AepUITemplateType.SMALL_IMAGE.typeName -> {
```

Comment:
Can we move this check to L100, before reading the card fields? title, body, image etc. are specific to SmallImageTemplate and should be read based on conforming template type

Reply:
Moved the check, and parsing based on template is moved to a separate function to keep this function body small.